app/audit/models/ManageYourDetailsResponseAuditModel.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package audit.models
+
+import audit.Utilities
+import auth.MtdItUser
+import enums.IncomeSourceJourney.{ForeignProperty, SelfEmployment, UkProperty}
+import models.incomeSourceDetails.viewmodels.ManageIncomeSourceDetailsViewModel
+import play.api.libs.json.{JsValue, Json}
+
+case class ManageYourDetailsResponseAuditModel(
+                                                viewModel: ManageIncomeSourceDetailsViewModel
+                                              )(implicit user: MtdItUser[_]) extends ExtendedAuditModel {
+  override val transactionName: String = enums.TransactionName.ManageIncomeSourceDetails
+  override val auditType: String = enums.AuditType.ManageIncomeSourceDetails
+
+  override val detail: JsValue =
+    Utilities.userAuditDetails(user) ++
+      Json.obj(
+        "journeyType" ->
+          (viewModel.incomeSourceType match {
+            case SelfEmployment => "SE"
+            case UkProperty => "UKPROPERTY"
+            case ForeignProperty => "FOREIGNPROPERTY"
+          }),
+        "businessName" -> (viewModel.tradingName match {
+          case Some(name) => s"$name"
+          case None => None
+        }),
+        "businessAddressLine1" -> viewModel.address.map(address => address.addressLine1),
+        "businessAddressLine2" -> viewModel.address.flatMap(address => address.addressLine2),
+        "businessAddressLine3" -> viewModel.address.flatMap(address => address.addressLine3),
+        "businessAddressTownOrCity" -> viewModel.address.flatMap(address => address.addressLine4),
+        "businessAddressPostcode" -> viewModel.address.flatMap(address => address.postCode),
+        "businessAddressCountry" -> viewModel.address.map(address => address.countryCode),
+        "dateStarted" -> viewModel.tradingStartDate,
+        "accountingMethod" ->
+          (viewModel.businessAccountingMethod match {
+            case Some(value) => if (value) {
```

Comment:
I would  believe you are extracting boolean value here, there is a easy way of doing this, without extra if/else:

```
viewModel.businessAccountingMethod match {
            case Some(true) => "Traditional accounting"
            case Some(false) => "Cash basis accounting"
            case None => None
          }
```